### PR TITLE
Rendered examples portal for various routes

### LINKED ISSUE #197: Show /examples popup when no example link provided
Instead of displaying a blank page for the following links:

https://vega.github.io/editor/#/examples

https://vega.github.io/editor/#/examples/vega

https://vega.github.io/editor/#/examples/vega-lite

Display the Vega Examples gallery popup you get when clicking on Examples

### PR DESCRIPTION
**Pull request by ganeshpatro321**
This PR will resolve #197.
Please review it.
And we can also add other possible routes, which should render the example portal.

Thank You.

### CODE REVIEW

File: src/components/app.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ import VizPane from './viz-pane';
 
 type Props = ReturnType<typeof mapDispatchToProps>;
 
-class App extends React.Component<Props & { match: any; location: any }> {
+class App extends React.Component<Props & { match: any; location: any; showExample?: any }> {
```
Comment (domoritz) [inline src/components/app.tsx]: This should be a boolean

File: src/components/app.tsx [hunk 2]
```diff
@@ -116,10 +116,11 @@ class App extends React.Component<Props & { match: any; location: any }> {
   }
 
   public render() {
+    const showExample = this.props.showExample ? true : false;
```
Comment (domoritz) [inline src/components/app.tsx]: Why?

File: src/components/app.tsx (hunk 1 above)
Comment (ganeshpatro321) [reply src/components/app.tsx]: Okay I will update it

File: src/components/app.tsx (hunk 2 above)
Comment (ganeshpatro321) [reply src/components/app.tsx]: I just added it for better understanding, we can remove it not needed.

Comment (domoritz) [reply src/components/app.tsx]: How does this help understanding to add a ternary that just copies the variable?

Comment (ganeshpatro321) [reply src/components/app.tsx]: Okay, yeah that doesnt make much sense, I am removing it

Comment (domoritz) [reply src/components/app.tsx]: You can keep the temp variable if you want.

File: src/components/app.tsx (hunk 1 above)
Comment (domoritz) [reply src/components/app.tsx]: Thanks.

**Comment by ganeshpatro321** [comment]
Check this, @domoritz

File: src/components/app-shell.tsx [hunk 3]
```diff
@@ -20,6 +20,9 @@ export default class AppShell extends React.Component<Props> {
           <Route path="/edited" component={App} />
           <Route path="/gist/:mode/:username/:id/:revision/:filename" component={App} />
           <Route path="/examples/:mode/:example_name" component={App} />
+          <Route path="/examples" component={() => <App showExample={true} />} />
+          <Route path="/examples/vega" component={() => <App showExample={true} />} />
+          <Route path="/examples/vega-lite" component={() => <App showExample={true} />} />
```
[the remark this answers is not in the document]
Comment (domoritz) [reply src/components/app-shell.tsx]: Actually, it does not work. http://0.0.0.0:8080/#/examples/vega and http://0.0.0.0:8080/#/examples/vega-lite should show a different tab in the modal.

**Comment by ganeshpatro321** [comment]
@domoritz The tab rendering is fixed. I think its good to go now.
Thank You!

**Comment by domoritz** [comment]
Thanks so much!

**Pull request merged**